File: Codigos/C++/C2/Listas e Trabalhos/PRODUTO/Produto/quimico.cpp
#include "quimico.h"

Quimico::Quimico() : Produto()
{
    this->inflamavel = false;
}

Quimico::Quimico(int codigo) : Produto(codigo)
{
    this->inflamavel = false;
}

Quimico::Quimico(int codigo, string *nome, int qnt, double valor) : Produto(codigo, nome, qnt, valor)
{
    this->inflamavel = false;
}

bool Quimico::getInflamavel() const
{
    return inflamavel;
}

void Quimico::setInflamavel(bool value)
{
    inflamavel = value;
}

string *Quimico::toString()
{
    char buffer[258] = {0};

    sprintf(buffer, "(%d):\t nome = %s\t inflamavel = %s\t valor = %.2lf\t qnt = %d\n",
           this->getCodigo(), this->getNome()->c_str(), this->inflamavel ? "sim" : "Nao" ,this->getValor(), this->getQnt());

    string *strBuffer = new string (buffer);

    return strBuffer;

}

Quimico::~Quimico()
{

}

File: Codigos/Codigos/JAVA/WorkSpace C2/Fabrica/src/model/Quimico.java
package model;

public class Quimico extends Produto {
	
	private boolean inflamavel;

	public Quimico() {
		this.inflamavel = false;
	}

	public Quimico(int codigo) {
		super(codigo);
		this.inflamavel = false;
	}

	public Quimico(int codigo, String nome, int quantidade, double preco) {
		super(codigo, nome, quantidade, preco);
		this.inflamavel = false;
	}

}
